packages/canary-web-components/src/components/ic-date-picker/ic-date-picker.tsx

```diff
@@ -0,0 +1,1512 @@
+
+  private isCurrentMonth = (): boolean => {
+    const d = new Date();
+    return (
+      d.getFullYear() === this.yearInView && d.getMonth() === this.monthInView
+    );
+  };
+
+  private updateMonthInView = (): void => {
+    this.currMonthView = this.getMonthView(this.focussedDate);
+
+    this.focussedDay = this.focussedDate.getDate();
+    this.monthInView = this.focussedDate.getMonth();
+    this.yearInView = this.focussedDate.getFullYear();
+
+    if (this.dayPickerKeyboardNav) {
+      this.monthInViewUpdateHandled = true;
+      this.setAriaLiveRegionText(this.getMonthInViewText());
+      this.dayPickerKeyboardNav = false;
+    }
+  };
+
+  private handleSelectDay = (day: Date): void => {
+    this.setSelectedDate(day);
+    this.calendarOpen = false;
+    this.inputEl.setCalendarFocus();
+  };
+
+  private handleSelectMonth = (month: number): void => {
+    this.moveMonths(month - this.monthInView);
+    this.monthPickerVisible = false;
+    this.setMonthSelectedLiveRegionText();
+
+    setTimeout(() => this.monthButtonEl.setFocus(), FOCUS_TIMER);
+  };
+
+  private handleSelectYear = (year: number): void => {
+    const yrPos = this.decadeView.indexOf(year);
+    if (yrPos > 0 && yrPos < this.decadeView.length - 1) {
+      this.moveYears(year - this.yearInView);
+      this.focusDay = false;
+      this.yearPickerVisible = false;
+      const monthName = this.monthNames[this.monthInView];
+      this.setAriaLiveRegionText(
+        `${year} selected. ${monthName} ${year} currently in view.`
+      );
+      setTimeout(() => this.yearButtonEl.setFocus(), FOCUS_TIMER);
+    } else {
+      const moveYears = year - this.focussedYear > 0 ? 10 : -10;
+      this.updateFocussedYear(moveYears, this.yearButtonFocussed);
+      this.setAriaLiveRegionText(this.getDecadeInViewText());
+    }
+  };
+
+  private monthPickerKeyDownHandler = (ev: KeyboardEvent): void => {
+    let handled = true;
+    switch (ev.key) {
+      case "ArrowUp":
+        this.updateFocussedMonth(-1);
+        break;
+
+      case "ArrowDown":
+        this.updateFocussedMonth(1);
+        break;
+
+      case "ArrowLeft":
+        this.updateFocussedMonth(-1);
+        break;
+
+      case "ArrowRight":
+        this.updateFocussedMonth(1);
+        break;
+
+      case "Home":
+        this.updateFocussedMonth(-this.focussedMonth);
+        break;
+
+      case "End":
+        this.updateFocussedMonth(11 - this.focussedMonth);
+        break;
+
+      case "Tab":
+        handled = this.calendarTabHandler(ev);
+        break;
+
+      case "Escape":
+        ev.stopImmediatePropagation();
+        this.monthPickerVisible = false;
+        setTimeout(() => this.focusFocussedDay(), FOCUS_TIMER);
+        break;
+
+      default:
+        handled = false;
+    }
+
+    if (handled) {
+      ev.preventDefault();
+    }
+  };
+
+  private yearPickerKeyDownHandler = (ev: KeyboardEvent): void => {
+    let handled = true;
+    switch (ev.key) {
+      case "ArrowUp":
```

Comment:
Same question here as the month picker key down handler - can some of the cases be merged?